docs/templating/layout/customize_storefront_layout.md

```diff
@@ -67,6 +67,130 @@ The most important templates are:
 |`cart/component/add_to_cart/add_to_cart.html.twig`|"add to cart" element|
 |`cart/component/summary/summary.html.twig`|cart summary|
 
+#### Extend twig template
+
+```html+twig
+{% extends '@IbexaCart/themes/standard/cart/component/minicart/minicart.html.twig' %}
+
+{% block content %}
+    <img
+        class="ibexa-store-minicart__image"
+        src="{{ asset('bundles/ibexastorefront/img/icons/cart.svg') }}"
+        alt="{{ 'minicart.icon.alt'|trans|desc('Cart') }}"
+    />
+    {{ parent() }}
+{% endblock %}
+```
+
+To avoid self-reference, ``@IbexaCart` is used instead of `@ibexadesign`.
+
+Components out-of-the-box are not styled as you can freely customize them according to your needs.
+CSS classes can be easily added like every element from the base Twig which has its own attr object.
+For example, if you want to add custom CSS classes to quantity input in Add to Cart component, use the following:
+
+```html+twig
+{% set quantity_input_attr = {
+    class: 'ibexa-store-input ibexa-store-input--number ibexa-store-add-to-cart__quantity-input',
+} %}
+```
+
+Every element is also inside its own block so that it is easy to override it.
+
+#### Extending JavaScript
+
+In case of the JavaScript component, you should import original class and extend it:
+
+```js
+import Minicart from '@ibexa-cart/src/bundle/Resources/public/js/component/minicart';
+
+export default class StorefrontMinicart extends Minicart {}
+```
+
+The example below shows how to add a clear button support to the maincart:
+
+```js
+import Maincart from '@ibexa-cart/src/bundle/Resources/public/js/component/maincart';
+
+export default class StorefrontMaincart extends Maincart {
+    constructor(options) {
+        super(options);
+
+        this.clearCartBtn = this.container.querySelector('.ibexa-store-maincart__clear-cart-btn');
+
+        this.onCartClear = this.onCartClear.bind(this);
+    }
+
+    attachStorefrontMaincartListeners() {
+        this.clearCartBtn.addEventListener('click', this.onCartClear, false);
+    }
+
+    onCartClear() {
+        this.cart.empty();
+    }
+}
+```
+
+### Main cart
+
+The base widget for the main cart view must be customized as out-of-the-box it consists only of the container with items. Each item consists of `<div>` wrappers with quantity input and remove item button. With customization you can add layout containers and items' data such as title, or price.
```

Comment:
```suggestion
You must customize the base widget for the main cart view, because out-of-the-box it consists only of the container with items.
Each item consists of `<div>` wrappers with quantity input and remove item button. With customization you can add layout containers and items' data such as title or price.
```